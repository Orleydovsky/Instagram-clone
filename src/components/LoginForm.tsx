import { signInWithEmailAndPassword } from 'firebase/auth'
import React, { useEffect, useState } from 'react'
import { Link } from 'react-router-dom'
import instagram from '../assets/instagram.svg'
import { auth } from '../services/firebase/firebase-config'
import { Input } from './Lib'

export interface formInputs {
  email: string,
  password: string,
  username?: string,
  name?: string
}

export function LoginForm () {
  useEffect(() => {
    document.title = 'Login â€¢ Instagram'
  })
  const [error, setError] = useState<string>()
  const [loading, setLoading] = useState(false)
  const [formInputs, setFormInputs] = useState<formInputs>({
    email: '',
    password: ''
  })
  const { email, password } = formInputs
  const isSubmitInvalid = email === '' || password === ''
  const handleFormInputs = (event: React.ChangeEvent<HTMLFormElement>) => {
    setFormInputs({
      ...formInputs,
      [event.target.name]: event.target.value
    })
  }
  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    try {
      setLoading(true)
      await signInWithEmailAndPassword(auth, email, password)
    } catch (error) {
      error instanceof Error ? setError(error.message) : console.error(error)
    } finally {
      setLoading(false)
    }
  }
  return (
    <div className='py-12 w-80'>
      <div className='bg-white justify-center flex flex-col items-center border border-gray-200 py-5'>
        <div className='flex justify-center my-6'>
            <img src={instagram} alt='Instagram logo' className='h-12' />
        </div>
          <form onChange={handleFormInputs} onSubmit={handleSubmit} className='w-64 flex flex-col'>
            <Input name='email' type='text' placeholder='Email address'/>
            <Input name='password' type='password' placeholder='Password'/>
            <button disabled={isSubmitInvalid} className='bg-blue-500 text-white rounded-md py-1 my-2 font-medium disabled:opacity-50'>
            {loading ? 'Loading...' : 'Login'}
            </button>
          </form>
      <div className='font-thin text-red-500 mt-5'>
      {error}
      </div>
      </div>
      <div className='bg-white my-3 text-center p-5 border border-gray-200'>
        Don't have an account? <Link to='/signup' className='text-blue-500 font-medium'>Sign up</Link>
      </div>
    </div>
  )
}
