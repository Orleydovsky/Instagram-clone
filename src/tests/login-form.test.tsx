import renderer from 'react-test-renderer'
import { it, expect } from 'vitest'
import { BrowserRouter as Router } from 'react-router-dom'
import LoginForm from '../components/login-form'

it('renders LoginForm', () => {
  const tree = renderer
    .create(
      <Router>
        <LoginForm/>
      </Router>
    )
    .toJSON()
  expect(tree).matchSnapshot()
})
