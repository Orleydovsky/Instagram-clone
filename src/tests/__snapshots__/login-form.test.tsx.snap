// Vitest Snapshot v1

exports[`renders LoginForm 1`] = `
<div
  className="py-12 w-80"
>
  <div
    className="bg-white justify-center flex flex-col items-center border border-gray-200 py-5"
  >
    <div
      className="flex justify-center my-6"
    >
      <img
        alt="Instagram logo"
        className="h-12"
        src="/src/assets/instagram.svg"
      />
    </div>
    <form
      className="w-64 flex flex-col"
      onSubmit={[Function]}
    >
      <input
        className="bg-gray-50 border w-full border-gray-200 rounded-sm p-2 text-sm my-1 box-border"
        name="email"
        onChange={[Function]}
        placeholder="Email address"
        type="text"
      />
      <input
        className="bg-gray-50 border w-full border-gray-200 rounded-sm p-2 text-sm my-1 box-border"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
      <button
        className="bg-blue-500 text-white rounded-md py-1 my-2 font-medium disabled:opacity-50"
        disabled={true}
      >
        Login
      </button>
    </form>
    <div
      className="font-thin text-red-500 mt-5"
    />
  </div>
  <div
    className="bg-white my-3 text-center p-5 border border-gray-200"
  >
    Don't have an account? 
    <a
      className="text-blue-500 font-medium"
      href="/signup"
      onClick={[Function]}
    >
      Sign up
    </a>
  </div>
</div>
`;
